lib/web_ui/lib/src/engine/dom.dart

```diff
@@ -626,6 +626,10 @@ extension DomCanvasRenderingContext2DExtension on DomCanvasRenderingContext2D {
 @staticInterop
 class DomImageData {}
 
+DomImageData createDomImageData(Object? dataOrSw, int shOrSw, [int? sh]) =>
```

Comment:
After some head scratching I see what's going on here but it's really confusing/difficult to read. Would it be possible to just explicitly have two different functions that cover the case where you have data vs not? Like `createDomImageData` and `createEmptyDomImageData`? think the weirdness of the parameter typing would be a lot less confusing that way.

Reply:
Yea, good call. I just copied it from `dart:html` blindly` but it makes sense to have two APIs.